x-pack/plugins/enterprise_search/server/lib/ml/start_ml_model_deployment.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { MlStartTrainedModelDeploymentRequest } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+
+import { MlTrainedModels } from '@kbn/ml-plugin/server';
+import { MlClient } from '@kbn/ml-plugin/server/lib/ml_client';
+
+import { MlModelDeploymentStatus, MlModelDeploymentState } from '../../../common/types/ml';
+
+import {
+  ElasticsearchResponseError,
+  isResourceNotFoundException,
+} from '../../utils/identify_exceptions';
+
+import { getMlModelDeploymentStatus } from './get_ml_model_deployment_status';
+
+export const acceptableModelNames = ['.elser_model_1_SNAPSHOT'];
+
+export const startMlModelDeployment = async (
+  modelName: string,
+  mlClient: MlClient | undefined,
+  trainedModelsProvider: MlTrainedModels | undefined
+): Promise<MlModelDeploymentStatus> => {
+  if (!trainedModelsProvider || !mlClient) {
+    throw new Error('Machine Learning is not enabled');
+  }
+
+  let deploymentState: MlModelDeploymentState = MlModelDeploymentState.NotDeployed;
```

Comment:
nitpick: you can just remove this initial assigment, it's only assigned once below.

Reply:
Good catch - this was from when the code was doing a lot more before a couple of commits ago...